### ISSUE TITLE
'num_return_sequences' Parameter for Text2Text generation Models

### ISSUE THREAD

**Issue by sharaku17** [huggingface/huggingface_hub#565]
**Is your feature request related to a problem? Please describe.**
I am trying to use the Inference API for Text2text generation Models, namely for Paraphraser Models. One example would be the Pegasus Paraphraser. Using this model in a colab with the transformers library works great, I can define the `num_return_sequences` Parameter and the Model outputs as many paraphrases as I want for a given input.

Now using this model with the inference API is where the problem appears, the `num_return_sequences` Parameter is ignored when using the inference API and I always get a single paraphrase as a response back. 

**Describe alternatives you've considered**
I tested other paraphrasing text2text models and this problem seems to involve most of them. I would be happy if it worked with the pegasus model as it outputs high-quality paraphrases during my testing in colab.

**Additional context**
If there is no imminent Fix possible, I would greatly appreciate how I could solve this problem on my own maybe. I am trying to build a very small web app that allows paraphrasing, this app will be only a personal project, so deploying such a model looks like overkill for my use case, which is why using the inference API looked like the perfect solution for me.

EDIT:
Further testing showed the same behavior when using pipelines. 
```
from transformers import pipeline
generator= pipeline('text2text-generation', 'tuner007/pegasus_paraphrase')
output = generator("Which course should I take to get started in data science?", do_sample= False ,num_return_sequences=3, num_beams=3)
```
Result: `[{'generated_text': 'Which data science course should I take?'}]`

Tried it with `do_sample = True` aswell. 

After some further investigation I found this PR: [Link](https://github.com/huggingface/transformers/pull/14410)
which tried to fix this problem, but it seems like the code was not merged, not sure if tests failed or if there was another issue 
#

**Comment by Narsil** [comment]
Hi @sharaku17 ,

Indeed the PR only needed a small quality fix `make fixup`. https://github.com/huggingface/transformers/pull/14411 but original author never did it.

I resurrected the PR here: https://github.com/huggingface/transformers/pull/14988 will be supported on the API once it gets merged (or very soon after lets say)

**Comment by sharaku17** [comment]
Hey @Narsil ,
first of all thank you for resurrecting the PR and also adding the needed fix. 
I saw that the PR was merged about 3 days ago, so I tried to see if it worked now, but I am still only receiving one Paraphrase back from the API.

Code is still the same as the one above in my first post. Any idea why this is still happening?

**Comment by yananchen1989** [comment]
Face the same issue, version:4.15.0

**Comment by Narsil** [comment]
@yananchen1989 this hasn't been merged in 4.15, can you try loading from `master` ? It will be available in 4.16 though.

**Comment by Narsil** [comment]
I will close the issue, since the API already runs the necessary code.

Feel free to reopen if you think this is not the case.
